common/src/main/java/com/fasterxml/jackson/databind/introspect/GuiceInjectableValues.java

```diff
@@ -36,6 +37,17 @@ public Object findInjectableValue(
       Object valueId, DeserializationContext ctxt, BeanProperty forProperty, Object beanInstance
   )
   {
-    return injector.getInstance((Key) valueId);
+    if(Key.class.isAssignableFrom(valueId.getClass())){
```

Comment:
This was a weird error I was getting where sometimes beanInstance was being passed as String of the class rather than a Key of the class.